### PR TITLE
Ensure main scene is set when running on device

### LINKED ISSUE #26008: Button to run on Android device doesn't check if main scene is set
**Godot version:**
commit 4e0eb752c2bfd7aac05023af67cd368061012bab

**OS/device including version:**
Editor build is Windows 10 64bit build
Android compiled for armv7 running on a Xiaomi Pocophone F1

**Issue description:**
Godot crashes pretty much after starting. I can find the following in the logcat:
```
02-18 21:17:36.187  1697  4148 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.godot.testandroid/org.godotengine.godot.Godot bnds=[258,696][426,864] (has extras)} from uid 10033
02-18 21:17:36.194   682   682 E ANDR-IOP: IOP HAL: Received pkg_name = com.godot.testandroid pid = -1
02-18 21:17:36.200  1697  4148 D ActivityTrigger: activityStartTrigger: Activity is Triggerred in full screen ApplicationInfo{57aaacb com.godot.testandroid}
02-18 21:17:36.200  1697  4148 E ActivityTrigger: activityStartTrigger: not whiteListedcom.godot.testandroid/org.godotengine.godot.Godot/1
02-18 21:17:36.201  1697  4148 D ActivityTrigger: activityResumeTrigger: The activity in ApplicationInfo{57aaacb com.godot.testandroid} is now in focus and seems to be in full-screen mode
02-18 21:17:36.201  1697  4148 E ActivityTrigger: activityResumeTrigger: not whiteListedcom.godot.testandroid/org.godotengine.godot.Godot/1
02-18 21:17:36.208  1697  3285 D ActivityTrigger: activityResumeTrigger: The activity in ApplicationInfo{57aaacb com.godot.testandroid} is now in focus and seems to be in full-screen mode
02-18 21:17:36.208  1697  3285 E ActivityTrigger: activityResumeTrigger: not whiteListedcom.godot.testandroid/org.godotengine.godot.Godot/1
02-18 21:17:36.219  1697  3285 D Boost   : hostingType=activity, hostingName=com.godot.testandroid/org.godotengine.godot.Godot, callerPackage=com.mi.android.globallauncher, isSystem=true, isBoostNeeded=false.
02-18 21:17:36.220  1697  3285 I ActivityManager: Start proc 22046:com.godot.testandroid/u0a186 for activity com.godot.testandroid/org.godotengine.godot.Godot caller=com.mi.android.globallauncher
02-18 21:17:36.220  4246  4531 D PowerKeeper.Event: notifyAMProcStart processName: 10186 reason: com.godot.testandroid, pid:0
02-18 21:17:36.292  4587 27260 D GameBoosterService: onGameStatusChange foreground:ForegroundInfo{mForegroundPackageName='com.godot.testandroid', mForegroundUid=10186, mForegroundPid=22046, mLastForegroundPackageName='com.mi.android.globallauncher', mLastForegroundUid=10033, mLastForegroundPid=28595, mMultiWindowForegroundPackageName='null', mMultiWindowForegroundUid=-1, mFlags=1}
--------- beginning of crash
02-18 21:17:36.515 22086 22086 F DEBUG   : pid: 22046, tid: 22081, name: GLThread 1416  >>> com.godot.testandroid <<<
02-18 21:17:36.523 22086 22086 F DEBUG   :     #04 pc 01a762f8  /data/app/com.godot.testandroid-39a4FJJwHH34IixB0nmOIA==/lib/arm/libgodot_android.so
02-18 21:17:36.523 22086 22086 F DEBUG   :     #05 pc 000a2490  /data/app/com.godot.testandroid-39a4FJJwHH34IixB0nmOIA==/lib/arm/libgodot_android.so (Java_org_godotengine_godot_GodotLib_getGlobal+80)
02-18 21:17:36.523 22086 22086 F DEBUG   :     #14 pc 00078ffa  /dev/ashmem/dalvik-classes.dex extracted in memory from /data/app/com.godot.testandroid-39a4FJJwHH34IixB0nmOIA==/base.apk (deleted) (org.godotengine.godot.Godot$2.run+22)
02-18 21:17:36.706  1697 22091 W ActivityManager:   Force finishing activity com.godot.testandroid/org.godotengine.godot.Godot
02-18 21:17:36.732  1697  2102 W InputDispatcher: channel 'e5b397b com.godot.testandroid/org.godotengine.godot.Godot (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
02-18 21:17:36.732  1697  2102 E InputDispatcher: channel 'e5b397b com.godot.testandroid/org.godotengine.godot.Godot (server)' ~ Channel is unrecoverably broken and will be disposed!
02-18 21:17:36.733  1697  2075 I ActivityManager: Process com.godot.testandroid (pid 22046) has died: vis  +99TOP
02-18 21:17:36.735  4246  4531 D PowerKeeper.Event: notifyAMProcDied pacakageName: com.godot.testandroid, pid:22046
02-18 21:17:36.742  1697  4161 I WindowManager: WIN DEATH: Window{e5b397b u0 com.godot.testandroid/org.godotengine.godot.Godot}
02-18 21:17:36.742  1697  4161 W InputDispatcher: Attempted to unregister already unregistered input channel 'e5b397b com.godot.testandroid/org.godotengine.godot.Godot (server)'
02-18 21:17:36.787  4587 27260 D GameBoosterService: onGameStatusChange foreground:ForegroundInfo{mForegroundPackageName='com.mi.android.globallauncher', mForegroundUid=10033, mForegroundPid=28595, mLastForegroundPackageName='com.godot.testandroid', mLastForegroundUid=10186, mLastForegroundPid=22046, mMultiWindowForegroundPackageName='null', mMultiWindowForegroundUid=-1, mFlags=0}
02-18 21:17:36.807   712  3585 W SurfaceFlinger: Attempting to destroy on removed layer: AppWindowToken{11bec7f token=Token{85b899e ActivityRecord{716fbd9 u0 com.godot.testandroid/org.godotengine.godot.Godot t232}}}#0
```

**Steps to reproduce:**
Create a simple project in Godot, I just have a 2D node with a button saved in a scene, then setup export to android and run.

Using GLES2 renderer, same issue with GLES3

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #26008

I wasn't really sure how to determine whether the editor checks scene from normal run or native run, so I went with a meta >.>

### CODE REVIEW

**Comment by reduz** [comment]
It should not even run if no scene is selected I think.

**Comment by KoBeWi** [comment]
That's what this PR does. If you don't set main scene, the "run on device" button behaves the same as running the project - it prompts you to select main scene.

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -1265,7 +1265,12 @@ void EditorNode::_dialog_action(String p_file) {
 			ProjectSettings::get_singleton()->set("application/run/main_scene", p_file);
 			ProjectSettings::get_singleton()->save();
 			//would be nice to show the project manager opened with the highlighted field..
-			_run(false, ""); // automatically run the project
+
+			if ((bool)pick_main_scene->get_meta("from_native")) {
```
Comment (akien-mga) [inline editor/editor_node.cpp]: Shouldn't you call `has_meta("from_native")` first?

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -4001,6 +3985,37 @@ bool EditorNode::has_scenes_in_session() {
 	return !scenes.empty();
 }
 
+bool EditorNode::ensure_main_scene(bool from_native) {
```
Comment (akien-mga) [inline editor/editor_node.cpp]: Parameter should be `p_from_native`.

File: editor/editor_node.cpp (hunk 1 above)
Comment (KoBeWi) [reply editor/editor_node.cpp]: Fixed.

File: editor/editor_node.cpp (hunk 2 above)
Comment (KoBeWi) [reply editor/editor_node.cpp]: Fixed.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**